lib/libc/stdlib/lib_mktemp.c

```diff
@@ -71,26 +59,66 @@
  ****************************************************************************/
 
 /****************************************************************************
- * Public Functions
+ * Public Function Prototypes
  ****************************************************************************/
 
+char *__randname(char *, uint8_t);
+
 /****************************************************************************
  * Name: mktemp
  *
  * Description:
- *   The use of mktemp is dangerous; use mkstemp instead.
+ *   The mktemp() function generates a unique temporary filename from template.
```

Comment:
This description looks similar with mkdtemp which is a POSIX API.
```
The mkdtemp() function shall create a directory with a unique name derived from template. The application shall ensure that the string provided in template is a pathname ending with at least six trailing 'X' characters. The mkdtemp() function shall modify the contents of template by replacing six or more 'X' characters at the end of the pathname with the same number of characters from the portable filename character set. The characters shall be chosen such that the resulting pathname does not duplicate the name of an existing file at the time of the call to mkdtemp(). The mkdtemp() function shall use the resulting pathname to create the new directory as if by a call to:
```
Is this mktemp right?

Reply:
Part of description might be similar but mkdtemp() creates a directory with unique name whereas mktemp () generates unique temporary filename.

Reply:
ok. thank you for explain.